### PR TITLE
build: reinstall packages script

### PR DESCRIPTION
**Pull request by mistermalm**
## **Describe pull-request**  
Automated process: 
cd into all packages, remove node_modules and package-lock.json. Then install dependencies again.

## **Issue Linking:**  
_Choose one of the following options_
- **Jira:** `CDEP-3501` [tegel-script-to-improve-debugging](https://tegel.atlassian.net/browse/CDEP-3501)    

## **How to test**  
_Provide detailed steps for testing, including any necessary setup._
1. cd to tegel directory
2. run command: `npm run reinstall`

## **Checklist before submission**
- [ ] All existing tests pass
- [ ] `npm run build-all` without errors

## **Additional context**  
Another suggestion is to split the script in two, one command to delete files, and one to install. Let me know what you think

### CODE REVIEW

File: reinstall.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
```
Comment (timrombergjakobsson) [inline reinstall.sh:2]: We could add `set -e` at the top to ensure the script stops if a command fails, preventing half-completed installations.
```
# Exit immediately if a command exits with a non-zero status
set -e
```
Not a must just a suggestion :)

File: reinstall.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+echo "installing dependencies in subdirectories"
+
+for dir in "${directories[@]}"; do
+  if [ -d "$dir" ]; then
+    echo "installing dependencies in $dir"
+    (cd $dir && npm i)
+  else
+    echo "Directory $dir does not exist"
+  fi
+done
```
Comment (timrombergjakobsson) [inline reinstall.sh:39]: Maybe we can add a final confirmation message?

```
# Final confirmation message
echo "Installation process completed successfully!"
```

File: reinstall.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  "./packages/react"
+  # Add more directories as needed
+)
+
+echo "removing node modules and package-lock.json in $PWD"
+rm -rf node_modules
+rm -rf package-lock.json
+
+echo "removing node modules and package-lock.json in subdirectories"
+
```
Comment (timrombergjakobsson) [inline reinstall.sh:19]: Maybe add a cache cleaning before reinstalling? If we feel it is beneficial. 

```
# Clean npm cache
echo "Cleaning npm cache"
npm cache clean --force
```

File: reinstall.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# Array of directories to go into
+directories=(
```
Comment (timrombergjakobsson) [inline reinstall.sh:4]: 💯

File: reinstall.sh [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  "./packages/core"
+  "./packages/angular"
+  "./packages/angular/projects/components"
+  "./packages/angular-17"
+  "./packages/angular-17/projects/components"
+  "./packages/react"
+  # Add more directories as needed
+)
+
+echo "removing node modules and package-lock.json in $PWD"
```
Comment (timrombergjakobsson) [inline reinstall.sh:14]: We could group it into a function also:

```
# Function to remove node_modules and package-lock.json safely
clean_directory() {
  if [ -d "$1/node_modules" ]; then
    echo "Removing node_modules in $1"
    rm -rf "$1/node_modules"
  fi
  if [ -f "$1/package-lock.json" ]; then
    echo "Removing package-lock.json in $1"
    rm -rf "$1/package-lock.json"
  fi
```

Comment (timrombergjakobsson) [reply reinstall.sh:14]: And then we run it after the clean clean command:

```
# Clean root node_modules and package-lock.json
echo "Removing node_modules and package-lock.json in $PWD"
clean_directory "$PWD"

```

**Review by theJohnnyMe — APPROVED** [review]
Really nice work @mistermalm! It saves a lot of time... and nerves too ;)

**Review by nathalielindqvist — APPROVED** [review]
Works well!

**Pull request merged**